### PR TITLE
Handle unexpected exceptions in EnsembleEvaluator

### PR DESCRIPTION
**Pull request by BjarneHerland**
Like EvaluatorTracker this code also calls `_Monitor.track()` and is thus exposed to the same unexpected exceptions we handled in #3315.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: ert_shared/ensemble_evaluator/evaluator.py [hunk 1]
```diff
@@ -247,46 +242,69 @@ def stop(self):
         self._loop.call_soon_threadsafe(self._stop)
         self._ws_thread.join()
 
+    def _signal_cancel(self):
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/evaluator.py:245]: `_handle_cancel` as it doesn't signal anything, but rather responds to a signal?

File: ert_shared/ensemble_evaluator/evaluator.py [hunk 2]
```diff
@@ -247,46 +242,69 @@ def stop(self):
...
+                        unsuccessful_connection_attempts = 0
+
+                    logger.debug("Waiting for evaluator shutdown")
+                    break
+                except ConnectionClosedError as e:
+                    logger.debug(
+                        "Connection closed unexpectedly in "
+                        f"run_and_get_successful_realizations: {e}"
+                    )
+                    # The monitor went away...  what the h*** does that mean?
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/evaluator.py:276]: It means that something prevented the closing handshake. So anything, really. Server crash, connection termination, network issues, etc.

File: ert_shared/ensemble_evaluator/evaluator.py (hunk 1 above)
Comment (BjarneHerland) [reply ert_shared/ensemble_evaluator/evaluator.py:245]: I beg to differ...  😄   This code just forwards the desire to cancel evaluation, returning immediately (hence `_signal` ). I'm open to rename it but it does not primarily respond to a signal.

File: ert_shared/ensemble_evaluator/evaluator.py (hunk 2 above)
Comment (BjarneHerland) [reply ert_shared/ensemble_evaluator/evaluator.py:276]: 👍

**Review by jondequinor — CHANGES_REQUESTED** [review]
You need to amend your comment to something useful, at the very least to something without censored sacrilege naughty words.

**Comment by BjarneHerland** [comment]
Yeah...   forgot that part. Will do!

**Comment by BjarneHerland** [comment]
Messed up git-history...

File: ert_shared/ensemble_evaluator/evaluator.py [hunk 3]
```diff
@@ -70,6 +71,10 @@ def config(self):
     def ensemble(self):
         return self._ensemble
 
+    @property
+    def model_finished(self):
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/evaluator.py:75]: Might cause a lot of confusion

Comment (BjarneHerland) [reply ert_shared/ensemble_evaluator/evaluator.py:75]: This one is closed - see new PR

**Pull request closed without merging**